File: java/BinarySearch.java
package org.example.Impratev.Algorithm;

public class BinarySearch {
//time complexity O(log(n))
    public int binarySearch(int []arr,int value){

        int low =0;
        int high=arr.length-1;
        while (low<=high){
            int mid=(low+high)/2;
            if(arr[mid]==value){
                return  mid;
            } else if (arr[mid]<value) {
                low =mid+1;
            }
            else {
                high=mid-1;
            }
        }
        return -1;
    }
}

File: py/binarySearch.py
def binarySearch(arr,value,size):
    # time coplixty O(log n)
    high= size -1
    low=0
    while low<=high:
        mid = int((low+high)/2)
        if arr[mid]==value:
            return mid
        elif arr[mid]<value:
            low= mid+1
        else :
            high=mid-1
    
    return-1


print(binarySearch([11,33,41,62,85,90],85,6))